### PR TITLE
M0090IDE-4172 Document the new test capabilities of the SDK 6

### PR DESCRIPTION
**Pull request by tdelhomenie-microej**
https://docs.microej.com/en/feature-sdk6-testsuite/

### CODE REVIEW

File: SDK6UserGuide/testProject.rst [hunk 1]
```diff
@@ -5,48 +5,113 @@ Test a Project
...
 
-MicroEJ is compliant with a subset of JUnit version 4. 
-MicroEJ JUnit processor supports the following annotations: ``@After``,
+The SDK relies on `JUnit <https://junit.org/junit4/>`__, the most popular Java testing framework, to define and execute the tests.
+It supports a subset of JUnit 4, namely the annotations: ``@After``,
 ``@AfterClass``, ``@Before``, ``@BeforeClass``, ``@Ignore``, ``@Test``.
 
 Each test case entry point must be declared using the ``org.junit.Test`` annotation (``@Test`` before a method declaration). 
-Refer to JUnit documentation to get details on usage of other annotations.
+Refer to `JUnit documentation <https://junit.org/junit4/>`__ to get details on theusage of other annotations.
```
Comment (denisbedin) [inline SDK6UserGuide/testProject.rst:28]:
```suggestion
Refer to `JUnit documentation <https://junit.org/junit4/>`__ to get details on the usage of other annotations.
```

File: SDK6UserGuide/testProject.rst [hunk 2]
```diff
@@ -5,48 +5,113 @@ Test a Project
...
+            ...
+            dependencies {
+               implementation("ej.library.test:junit:1.7.1")
+            }
+            ...
+         }
+      }
+   }
+
+The test class can npw be created in the ``src/test/java`` folder.
```
Comment (denisbedin) [inline SDK6UserGuide/testProject.rst:113]:
```suggestion
The test class can now be created in the ``src/test/java`` folder.
```

File: SDK6UserGuide/testProject.rst [hunk 3]
```diff
@@ -60,106 +125,198 @@ or Application projects (plugin ``com.microej.gradle.application``).
       - right-click on the ``src/test/java`` folder.
       - select :guilabel:`New` > :guilabel:`Java Class``, then press ``Alt`` + ``Insert`` and select ``Test Method``.
```
Comment (denisbedin) [inline SDK6UserGuide/testProject.rst:126]:
```suggestion
      - select :guilabel:`New` > :guilabel:`Java Class`, then press ``Alt`` + ``Insert`` and select ``Test Method``.
```

File: SDK6UserGuide/testProject.rst (hunk 1 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/testProject.rst]: Done.

File: SDK6UserGuide/testProject.rst (hunk 2 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/testProject.rst]: Done.

File: SDK6UserGuide/testProject.rst (hunk 3 above)
Comment (tdelhomenie-microej) [reply SDK6UserGuide/testProject.rst]: Done.

**Review by denisbedin — APPROVED** [review]

**Pull request merged**